theseus/embodied/robot/joint.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+import warnings
+from typing import Optional
+import torch
+
+from theseus.geometry.functional import so3, se3
+
+
+class Joint(abc.ABC):
```

Comment:
Should we have any unit tests accompanying these or will that come in later PRs that add other data structures.